xmtp_id/src/associations/unverified.rs

```diff
@@ -344,14 +348,24 @@ pub struct UnverifiedSmartContractWalletSignature {
     pub(crate) signature_bytes: Vec<u8>,
     pub(crate) account_id: AccountId,
     pub(crate) block_number: u64,
+    pub(crate) chain_id: u64,
+    pub(crate) hash: [u8; 32],
 }
 
 impl UnverifiedSmartContractWalletSignature {
-    pub fn new(signature_bytes: Vec<u8>, account_id: AccountId, block_number: u64) -> Self {
+    pub fn new(
+        signature_bytes: Vec<u8>,
+        account_id: AccountId,
+        block_number: u64,
+        chain_id: u64,
+        hash: [u8; 32],
```

Comment:
None of the other unverified signature types store the hash, because the hash gets provided as part of signature verification (check out the `to_verified`) method of `IdentityUpdate`. 

We should really keep that consistent or it's going to cause confusion for devs.